packages/openai/src/openai-speech-model.ts

```diff
@@ -0,0 +1,139 @@
+import { SpeechModelV1, SpeechModelV1CallWarning } from '@ai-sdk/provider';
+import {
+  combineHeaders,
+  createBinaryResponseHandler,
+  parseProviderOptions,
+  postJsonToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { OpenAIConfig } from './openai-config';
+import { openaiFailedResponseHandler } from './openai-error';
+import { OpenAISpeechModelId } from './openai-speech-settings';
+import { OpenAISpeechAPITypes } from './openai-api-types';
+
+// https://platform.openai.com/docs/api-reference/audio/createSpeech
+const OpenAIProviderOptionsSchema = z.object({
+  instructions: z.string().nullish(),
+  speed: z.number().min(0.25).max(4.0).default(1.0).nullish(),
+});
+
+export type OpenAISpeechCallOptions = z.infer<
+  typeof OpenAIProviderOptionsSchema
+>;
+
+interface OpenAISpeechModelConfig extends OpenAIConfig {
+  _internal?: {
+    currentDate?: () => Date;
+  };
+}
+
+export class OpenAISpeechModel implements SpeechModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: OpenAISpeechModelId,
+    private readonly config: OpenAISpeechModelConfig,
+  ) {}
+
+  private getArgs({
+    text,
+    voice = 'alloy',
+    outputMediaType = 'mp3',
+    speed,
+    instructions,
+    providerOptions,
+  }: Parameters<SpeechModelV1['doGenerate']>[0]) {
+    const warnings: SpeechModelV1CallWarning[] = [];
+
+    // Parse provider options
+    const openAIOptions = parseProviderOptions({
+      provider: 'openai',
+      providerOptions,
+      schema: OpenAIProviderOptionsSchema,
+    });
+
+    let responseFormat = outputMediaType;
+
+    if (
+      outputMediaType &&
+      !['mp3', 'opus', 'aac', 'flac', 'wav', 'pcm'].includes(outputMediaType)
```

Comment:
what is the format for output media types? I was thinking it would be `audio/mp3` etc (IANA) - are we using something else here? or cutting the `audio/` part?